File: Frontend/students-employment-records-site/src/domain/students/enums/socialStatus.ts
import { enumToArrayNumber } from "@/tools/enums/enumUtils"
import { NeverUnreachable } from "@/tools/neverUreachable"

export enum SocialStatus {
    Orphan = 1,
    Invalid = 2,
    OVZ = 3,
}

export namespace SocialStatus {
    export function getAll(): SocialStatus[] {
        return enumToArrayNumber<SocialStatus>(SocialStatus)
    }

    export function getDisplayText(status: SocialStatus): string {
        switch (status) {
            case SocialStatus.Orphan:
                return "Сирота"
            case SocialStatus.Invalid:
                return "Инвалидность"
            case SocialStatus.OVZ:
                return "ОВЗ"
            default:
                throw new NeverUnreachable(status)
        }
    }
}

File: Backend/SER.Domain/Students/Enums/SocialStatus.cs
using System.Linq;

namespace SER.Domain.Students.Enums;
public enum SocialStatus
{
	Orphan = 1,
	Invalid = 2,
	OVZ = 3
}

public static class SocialStatusExtensions
{
	public static String DisplayName(this SocialStatus status)
	{
		return status switch
		{
			SocialStatus.Orphan => "Сирота",
			SocialStatus.Invalid => "Инвалидность",
			SocialStatus.OVZ => "ОВЗ",
			_ => throw new Exception()
		};
	}

	public static String DisplayName(this SocialStatus[] statuses)
	{
		if (statuses.Length == 0) return "-";

		String[] statusStrings = [.. statuses.Select(s => s.DisplayName())];
		return String.Join(", ", statusStrings);
	}
}
